data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaSinkProducer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.producer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSinkConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.TopicConfig;
+import org.opensearch.dataprepper.plugins.kafka.util.MessageFormat;
+
+import java.util.List;
+
+
+/**
+ * * A helper class which helps takes the buffer data
+ * and produce it to a given kafka topic
+ */
+
+@SuppressWarnings("deprecation")
+public class KafkaSinkProducer {
+
+    final KafkaProducer<String, Object> producer;
+
+    final KafkaSinkConfig kafkaSinkConfig;
+
+    final Record<Event> record;
+
+    final String schemaType;
+
+    public KafkaSinkProducer(final KafkaProducer producer,
+                             final Record<Event> record,
+                             final KafkaSinkConfig kafkaSinkConfig,
+                             final String schemaType) {
+        this.producer = producer;
+        this.record = record;
+        this.kafkaSinkConfig = kafkaSinkConfig;
+        this.schemaType = schemaType;
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public void produceRecords() {
+        if (MessageFormat.JSON.toString().equalsIgnoreCase(schemaType)) {
```

Comment:
Instead of if/else should we create separate serializer objects for each schema type and have them injected into this class?

Reply:
As the structure has been changed so  please review PR #2950.